src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -624,4 +627,31 @@ public void checkSkin(String adSkinLeftPath, String adSkinRightPath) {
     PageObjectLogging.log("SKIN", "SKIN presents on the page", true);
   }
 
+  /**
+   * Check if AdEngine loaded the ad web elements inside slot.
+   */
+  public boolean checkIfSlotOnPageLoaded(String slotName) {
+    changeImplicitWait(250, TimeUnit.MILLISECONDS);
+    try {
+      String slotSelector = AdsContent.getSlotSelector(slotName);
+      WebElement slot = driver.findElement(By.cssSelector(slotSelector));
+      if (slotName.equals(AdsContent.FLOATING_MEDREC)) {
+        triggerFloatingMedrec(slot);
+      }
+      List<WebElement> adWebElements = slot.findElements(By.cssSelector("div"));
+      return adWebElements.size() > 0;
+    } finally {
+      restoreDeaultImplicitWait();
+    }
+  }
+
+  private void triggerFloatingMedrec(WebElement slot) {
+    executeScript("window.scroll(0, 5000); setTimeout(function () {window.scroll(0, 5001) }, 500)");
+    // try to wait a floating medrec 3 seconds
+    try {
+      new WebDriverWait(driver, 3).until(CommonExpectedConditions.elementVisible(slot));
+    } catch (org.openqa.selenium.TimeoutException ignore) {
```

Comment:
The method tries to trigger a floating Medrec and waits 3 seconds till slot will be loaded.
I used `triggerFloatingMedrec` inside `isSlotOnPageLoaded` which can return false if a slot is not loaded. So, it could be the case when we try to trigger medrec, but configuration is not allow to do it. Should I rename to `tryTriggerFloatingMedrec`?